sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,509 @@
+
+    @distributed_trace
+    def update_entity(  # pylint:disable=R1710
+            self,
+            mode,  # type: Any
+            table_entity_properties,  # type: dict
+            partition_key=None,  # type: Optional[str]
+            row_key=None,  # type: Optional[str]
+            etag=None,   # type: Optional[object]
+            match_condition=None,  # type: Optional[MatchCondition]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Update entity in a table.
+
+        :param mode: Merge or Replace entity
+        :type mode: ~azure.table._models.UpdateMode
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, object]
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :param etag: Etag of the entity
+        :type etag: str
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
+
+        if mode is UpdateMode.replace:
+            try:
+                self._client.table.update_entity(
+                    table=self.table_name,
+                    partition_key=partition_key,
+                    row_key=row_key,
+                    table_entity_properties=table_entity_properties,
+                    if_match=if_match or if_not_match or "*",
+                    **kwargs)
+            except ResourceNotFoundError:
+                raise ResourceNotFoundError
+        if mode is UpdateMode.merge:
+            try:
+                self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
+                                                row_key=row_key, if_match=if_match or if_not_match or "*",
+                                                table_entity_properties=table_entity_properties, **kwargs)
+            except ResourceNotFoundError:
+                raise ResourceNotFoundError
+
+    @distributed_trace
+    def query_entities(
+            self,
+            query_options=None,  # type: Optional[QueryOptions]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> ItemPaged
```

Comment:
you should be able to put in brackets the type returned here like `ItemPaged[TableEntity]`. also in the docstring